kyaml/kio/byteio_reader.go

```diff
@@ -119,7 +124,22 @@ func (r *ByteReader) Read() ([]*yaml.RNode, error) {
 
 	index := 0
 	for i := range values {
-		decoder := yaml.NewDecoder(bytes.NewBufferString(values[i]))
+		value := values[i]
+
+		if r.Path != "" {
+			for _, g := range JSONMatch {
+				if match, err := filepath.Match(g, filepath.Ext(r.Path)); err != nil {
+					return nil, errors.Wrap(err)
+				} else if match {
+					value, err = yaml.ConvertJSONToYamlString(value)
```

Comment:
Is this necessary -- won't the yaml parser parse json?

Reply:
I tested it and it doesn't parse as expected. It adds escape characters etc. This way is more cleaner.

Reply:
What do you mean it adds escape characters?

https://github.com/kubernetes-sigs/kustomize/pull/2575

Reply:
@pwittrock It parses standalone json node but it has problem while wrapping it in a ResourceList. The output looks like this.

```
apiVersion: config.kubernetes.io/v1alpha1
kind: ResourceList
items:
- {"kind": "Deployment", "metadata": {"labels": {"app": "nginx2"}, "name": "foo",
    "annotations": {"app": "nginx2", config.kubernetes.io/index: '0', config.kubernetes.io/path: 'f1.json'}},
  "spec": {"replicas": 1}}
```

Instead, if we convert it to yaml string, rest of it will be taken care by existing logic. WDYT?